plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -403,6 +404,12 @@ public function log_errors() {
 				$context
 			);
 
+			// Record fatal error stats.
```

Comment:
I think we need to respect user's consent here? Since this is calling an external API 🤔

Reply:
You're right. I had that in the issue acceptance criteria but forgot to implement it. 😓  I'll update the code.

Reply:
It seems `McStats` is going to be used exclusively in `class-woocommerce.php` and does not require any constructor dependencies. I'm not against it, but I think we don't need to add this class to the container in this case.

Reply:
Actually, I think we will use McStats in other places in the future. It's better than using tracks to count events. 🙂

Reply:
Done, I added a check for user's consent before recording the stats in  fe088e72d572bbf840b60601a1c890d6ba3508bc

Reply:
@ilyasfoo I'm seeing the stats now. 😃 

1. Go to MC Stats
2. Click on `Simple List - active`
3. Search for `woocommerce-error`

Reply:
>Done, I added a check for user's consent before recording the stats in https://github.com/woocommerce/woocommerce/commit/fe088e72d572bbf840b60601a1c890d6ba3508bc

@chihsuan Nice, thanks! Consent check looks good